### PR TITLE
Require at least the versions that Sage uses in their own distribution

### LINKED ISSUE #14: Runtime dependencies should be at least  the version that Sage's SPKGs provide

### PR DESCRIPTION
**Pull request by saraedum**
fixes #14

### CODE REVIEW

**Comment by saraedum** [comment]
@isuruf: Is it a known problem that the CI builds *suceed* if they can not find one of the dependencies? (because of a type in the version number, e.g.)

**Comment by saraedum** [comment]
Hm, that's not very helpful…
```
Can't build /recipe_root due to unsatisfiable dependencies:
The following specifications were found to be in conflict:
  - sage 8.1 0
Use "conda info <package>" to see the dependencies for each package.
```

**Comment by saraedum** [comment]
Some linbox → givaro dependency seems to be wrong:
```
ImportError: /feedstock_root/build_artefacts/sage_1524308232972/_t_env/lib/python2.7/site-packages/sage/matrix/../../../../liblinboxsage.so.0: undefined symbol: _ZN6FFPACK7MinPolyIN6Givaro7ModularIddEESt6vectorIdSaIdEEEERT0_RKT_S8_mNS9_16ConstElement_ptrEmNS9_11Element_ptrEmPmNS_18FFPACK_MINPOLY_TAGEmmm
```

**Comment by saraedum** [comment]
@isuruf: Would you like to see any additional pins here?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -6,133 +6,139 @@ package:
...
+    # We require all dependencies with at least that version that upstream uses
+    # in their SPKGs. For packages that (seem to) use semantic versioning and
+    # are at least 1.0.0, we limit the version to the next major release.
+    - alabaster >=0.7.8
+    - appnope >=0.1.0  # [osx]
+    - arb >=2.11.1,<3
+    - backports.shutil_get_terminal_size >=1.0.0,<2
+    - backports_abc >=0.5
+    - bdw-gc >=7.2f,<8
+    - boost-cpp >=1.61.0,<2
```
Comment (isuruf) [inline recipe/meta.yaml]: @saraedum, what do you think about removing the pinnings for those that are already in `sagelib`? This looks weird.

**Pull request merged**